### PR TITLE
Add scaffold for sockets

### PR DESCRIPTION
**Pull request by Roller23**
Adds socket.io events for authentication and sending messages. There are example requests saved in our postman collection

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -284,6 +284,43 @@ async function generateNicknameTag(db, nickname) {
...
+         console.log('socket disconnected due to inactivity');
+         socket.disconnect(true);
+      }, 1000 * 10);
+
+      socket.on('disconnect', () => {
+         console.log('socket disconnected');
+      });
+      
+      socket.on('authenticate', async data => {
+         if (!data || !data.selector || !data.token) {
```
Comment (TinyRogue) [inline server.js:299]: Could be simplified to !data?.selector || !data?.token, coudn't be, huh?

File: server.js [hunk 2]
```diff
@@ -284,6 +284,43 @@ async function generateNicknameTag(db, nickname) {
...
+         if (!hash.verify(data.token, user.token)) {
+            console.log('hash not verified')
+            return socket.emit('auth denied');
+         }
+         socket._storage.user = user;
+         clearTimeout(socket._storage.timeout);
+         socket.emit('authenticated');
+         console.log(`${user.email} authenticated`);
+
+         socket.join('GENERAL_CHANNEL');
```
Comment (TinyRogue) [inline server.js:318]: WHAT IS GENERAL CHANNEL DUDE

File: server.js [hunk 3]
```diff
@@ -284,6 +284,43 @@ async function generateNicknameTag(db, nickname) {
...
+
+         socket.on('message', async data => {
+            console.log(`${socket._storage.user.nickname} sent "${data.message}"`);
+         });
+      });
    });
 
    const PORT = process.env.PORT || 3000;
```
Comment (TinyRogue) [inline server.js:326]: Everything is great, merge that guy

**Review by TinyRogue — APPROVED** [review]

File: server.js (hunk 1 above)
Comment (Roller23) [reply server.js:299]: yes

File: server.js (hunk 2 above)
Comment (Roller23) [reply server.js:318]: I DONT KNOW DUDE

**Pull request merged**